TFBarcodeScanner/TFBarcodeScannerViewController.m

```diff
@@ -239,21 +255,33 @@ - (void)setTorchOn:(BOOL)torchOn
 - (void)barcodePreviewWillShowWithDuration:(CGFloat)duration
 {
     // Subclasses can override
+    if (delegate != nil && [delegate respondsToSelector:@selector(barcodePreviewWillShowWithDuration:)]) {
```

Comment:
No reason to check if delegate is nil, since `respondsToSelector` will return false if the delegate is nil. And unless you add @optional to the delegate protocol (recommended), there is no reason to check with `respondsToSelector`. Same with the other delegate calls below.